### PR TITLE
feat: add postgres database deploy workflow

### PR DESCRIPTION
**Pull request by RMCampos**
# Description

Add image configuration, build steps and deploy workflows to have a postgres running every time a new pull request is open. 

PS: This change affects only `dev` environment. For `test` and `prod` will be in another PR.

Fixes [FSADT2-93](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-93)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

This was tested using other Pull Requests.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -104,6 +104,12 @@
 			<groupId>com.h2database</groupId>
```
Comment (paulushcgcj) [inline pom.xml:104]: why do you still have H2 database? If it's for testing purposes, you should scope it as test.

File: pom.xml [hunk 2]
```diff
@@ -104,6 +104,12 @@
...
 			<artifactId>h2</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+			<version>42.5.1</version>
```
Comment (paulushcgcj) [inline pom.xml:110]: You can get over without setting up the version, as spring controls driver versions.

**Review by DerekRoberts — APPROVED** [review]
Good work!  :)

**Review by paulushcgcj — CHANGES_REQUESTED** [review]
Pay attention to the compose file, as you're pushing images to github, and unless you are sure that who's running it got it's environment properly configured, or change it to include the github domain.

Also, the DatabaseDockerfile can be moved to a folder and the name can be changed to Dockerfile again.

**Comment by RMCampos** [comment]
@paulushcgcj thanks for the docker-compose file review. it was wrong indeed.

**Review by paulushcgcj — APPROVED** [review]

**Pull request merged**